File: Grace/brute_force/카펫.py
def solution(brown, yellow):
    answer = []
    
    # square = brown + yellow 
    # (yellow가로 * 2) + (yellow세로 * 2) + 4 = brown격자수
    
    x = 0
    y = 0
    
    for i in range(1, yellow + 1):
        if yellow % i == 0:
            x = yellow/i
            y = i
            if x * 2 + y * 2 + 4 == brown:
                answer.append(x + 2)
                answer.append(y + 2)
                answer.sort(reverse=True)
                break
    
    return answer

File: donggyun/brute_force/카펫.js
function solution(brown, yellow) {
  // yellow / i: 노란색 가로 길이, i: 노란색 세로 길이
  for (let i = 1; i <= Math.sqrt(yellow); i++) {
    const border = 2 * (yellow / i + 2) + 2 * i;
    if (yellow % i === 0 && brown === border) {
      return [yellow / i + 2, i + 2];
    }
  }
}

console.log(solution(24, 24));
